src/cursor/cur_std.c

```diff
@@ -1175,13 +1179,88 @@ __wt_cursor_bound(WT_CURSOR *cursor, const char *config)
         WT_ERR(__wt_config_gets(session, cfg, "bound", &cval));
         if (cval.len == 0)
             WT_ERR_MSG(session, EINVAL, "setting bounds must require the bound configuration set");
+
+        /* The cursor must have a key set to place the lower or upper bound. */
+        WT_ERR(__cursor_checkkey(cursor));
+
+        if (WT_STRING_MATCH("upper", cval.str, cval.len)) {
+            /*
+             * If the lower bounds are set, make sure that the upper bound is greater than the
+             lower
+             * bound.
+             */
+            WT_ERR(__wt_cursor_get_raw_key(cursor, &key));
+            if (F_ISSET(cursor, WT_CURSTD_BOUND_LOWER)) {
+                WT_ERR(__wt_compare(
+                  session, CUR2BT(cursor)->collator, &key, &cursor->lower_bound, &exact));
+                if (exact < 0)
+                    WT_ERR_MSG(session, EINVAL, "The upper bound overlaps with the lower bound");
+            }
+            /* Copy the key over to the upper bound item and set upper bound and inclusive flags. */
+            F_SET(cursor, WT_CURSTD_BOUND_UPPER);
+            WT_ERR(__wt_config_gets(session, cfg, "inclusive", &cval));
+            if (cval.val)
+                F_SET(cursor, WT_CURSTD_BOUND_LOWER_INCLUSIVE);
+            WT_ERR(__wt_buf_set(session, &cursor->upper_bound, key.data, key.size));
+
+            /* Clear the key the cursor was holding. */
+            F_CLR(cursor, WT_CURSTD_KEY_SET);
+            __wt_buf_free(session, &cursor->key);
+            WT_CLEAR(cursor->key);
+        } else if (WT_STRING_MATCH("lower", cval.str, cval.len)) {
+            /*
+             * If the upper bounds are set, make sure that the lower bound is less than the upper
+             * bound.
+             */
+            WT_ERR(__wt_cursor_get_raw_key(cursor, &key));
+            if (F_ISSET(cursor, WT_CURSTD_BOUND_UPPER)) {
+                WT_ERR(__wt_compare(
+                  session, CUR2BT(cursor)->collator, &key, &cursor->upper_bound, &exact));
+                if (exact > 0)
+                    WT_ERR_MSG(session, EINVAL, "The lower bound overlaps with the upper bound");
+            }
+            /* Copy the key over to the upper bound item and set upper bound and inclusive flags. */
+            F_SET(cursor, WT_CURSTD_BOUND_LOWER);
+            WT_ERR(__wt_config_gets(session, cfg, "inclusive", &cval));
+            if (cval.val)
```

Comment:
Can you check in GDB that by default inclusive is set to true and we pick it up as true here?

Reply:
Yep! Double checked that cval.val returns back 1 here
```
1219                WT_ERR(__wt_config_gets(session, cfg, "inclusive", &cval));
(gdb) 
1220                if (cval.val)
(gdb) p cval.val
$1 = 1
```